infrastructure/mail/mail.go

```diff
@@ -74,9 +75,12 @@ func NewMailer(config Config) (*Mailer, error) {
 
 	smtpClient, err := smtpServer.Connect()
 	if err != nil {
-		return &Mailer{nil, Defaults{}, false, config.DomainName}, err
+		log.Printf("mailer failed: %+v", err)
+		return &Mailer{nil, Defaults{}, false, config.DomainName}
```

Comment:
This risks panics if SMTP is misconfigured and we try to send an email - except that, rather than failing to start and making the misconfiguration obvious, it'd only crash after it has (potentially) been running for days.

Reply:
The mailer won't panic out, if it fails then it will just send a log message and if in the case of reset password, a link to the resetter to send to the user

Reply:
It still seems risky to store a `nil` pointer unless we're clearly documenting that it can be nil _and_ we're sure that anywhere that could use it is aware of this.

Reply:
I'll investigate this and report back

Reply:
I have changed it to instead of being a bool to check, it will go nil and validate it is not nil